src/factories/custom/Restricted_PIM_Factory_v1.sol

```diff
@@ -0,0 +1,365 @@
+            issuanceToken,
+            address(this) // assigns owner role to itself initially to manage minting rights temporarily
+        );
+
+        // set mint wrapper as minter
+        issuanceToken.setMinter(address(mintWrapper), true);
+        // MODIFY FUNDING MANAGER CONFIG
+        // decode configData of fundingManager
+        // store bcProperties and collateralToken to return
+        IFM_BC_Bancor_Redeeming_VirtualSupply_v1.BondingCurveProperties memory
+            bcProperties;
+        address collateralToken;
+        {
+            bytes memory fundingManagerConfigData =
+                fundingManagerConfig.configData;
+            (, bcProperties, collateralToken) = abi.decode(
+                fundingManagerConfigData,
+                (
+                    address,
+                    IFM_BC_Bancor_Redeeming_VirtualSupply_v1
+                        .BondingCurveProperties,
+                    address
+                )
+            );
+        }
+        // set newly deployed mint wrapper as issuance token
+        fundingManagerConfig.configData =
+            abi.encode(address(mintWrapper), bcProperties, collateralToken);
+
+        // deploy workflow
+        IOrchestrator_v1 orchestrator = IOrchestratorFactory_v1(
+            orchestratorFactory
+        ).createOrchestrator(
+            workflowConfig,
+            fundingManagerConfig,
+            authorizerConfig,
+            paymentProcessorConfig,
+            moduleConfigs
+        );
+
+        // return deployed contracts and all relevant decoded config params
+        return (
+            DeployedContracts({
+                issuanceToken: issuanceToken,
+                mintWrapper: mintWrapper,
+                orchestrator: orchestrator
+            }),
+            DecodedConfigParams({
+                collateralToken: IERC20(collateralToken),
+                initialCollateralSupply: bcProperties.initialCollateralSupply,
+                initialIssuanceSupply: bcProperties.initialIssuanceSupply,
+                realAdmin: realAdmin
+            })
+        );
+    }
+
+    function _configureWorkflow(
+        DeployedContracts memory deployedContracts,
+        DecodedConfigParams memory decodedConfigParams,
+        address beneficiary
+    ) private {
+        // get bonding curve / funding manager
+        address fundingManager =
+            address(deployedContracts.orchestrator.fundingManager());
+        // get payment router
+        address[] memory modules = deployedContracts.orchestrator.listModules();
+        address paymentRouter;
+        bytes4 paymentRouterInterfaceId =
+            type(ILM_PC_PaymentRouter_v1).interfaceId;
+        for (uint i; i < modules.length; ++i) {
+            if (
+                ERC165(modules[i]).supportsInterface(
+                    paymentRouterInterfaceId
+                )
+            ) {
+                paymentRouter = modules[i];
```

Comment:
Can break here

Reply:
This is fine as per https://github.com/InverterNetwork/contracts/pull/672#discussion_r1808532724

Reply:
Additional info: deploying is permissioned now